src/store.py

```diff
@@ -96,85 +107,132 @@ class FullNodeStore:
     async def _clear_database(self):
         await self.db.execute("DELETE FROM blocks")
         await self.db.execute("DELETE FROM potential_blocks")
+        await self.db.execute("DELETE FROM small_header_blocks")
         await self.db.commit()
 
     async def add_block(self, block: FullBlock) -> None:
-        await self.db.execute(
+        cursor_1 = await self.db.execute(
             "INSERT OR REPLACE INTO blocks VALUES(?, ?, ?)",
             (block.height, block.header_hash.hex(), bytes(block)),
         )
+        await cursor_1.close()
+        assert block.header_block.challenge is not None
+        small_header_block: SmallHeaderBlock = SmallHeaderBlock(
+            block.header_block.header, block.header_block.challenge
+        )
+        cursor_2 = await self.db.execute(
+            ("INSERT OR REPLACE INTO small_header_blocks VALUES(?, ?, ?, ?)"),
+            (
+                block.height,
+                block.header_hash.hex(),
+                bytes(block.header_block.proof_of_space.pool_pubkey).hex(),
+                bytes(small_header_block),
+            ),
+        )
+        await cursor_2.close()
         await self.db.commit()
 
     async def get_block(self, header_hash: bytes32) -> Optional[FullBlock]:
         cursor = await self.db.execute(
             "SELECT * from blocks WHERE header_hash=?", (header_hash.hex(),)
         )
         row = await cursor.fetchone()
+        await cursor.close()
         if row is not None:
             return FullBlock.from_bytes(row[2])
         return None
 
-    async def get_blocks(self) -> AsyncGenerator[FullBlock, None]:
-        async with self.db.execute("SELECT * FROM blocks") as cursor:
-            async for row in cursor:
-                yield FullBlock.from_bytes(row[2])
+    async def get_header_blocks_by_hash(
+        self, header_hashes: List[bytes32]
+    ) -> List[HeaderBlock]:
+        if len(header_hashes) == 0:
+            return []
+        header_hashes_db = tuple(h.hex() for h in header_hashes)
+        formatted_str = f'SELECT * from blocks WHERE header_hash in ({"?," * (len(header_hashes_db) - 1)}?)'
+        cursor = await self.db.execute(formatted_str, header_hashes_db)
+        rows = await cursor.fetchall()
+        await cursor.close()
+        header_blocks: List[HeaderBlock] = []
+        for row in rows:
+            header_blocks.append(FullBlock.from_bytes(row[2]).header_block)
+        return sorted(header_blocks, key=lambda hb: hb.height)
```

Comment:
Maybe it makes sense to return them in the order matching the passed list.

Reply:
Yeah good point.